java-squid/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -389,7 +389,17 @@ private Resolution findMethod(Env env, JavaType callSite, JavaType site, String
     return findMethod(env, callSite, site, name, argTypes, typeParams, false);
   }
 
+  private Resolution findConstructor(Env env, JavaType site, List<JavaType> argTypes, List<JavaType> typeParams, boolean autoboxing) {
+    return findMethod(env, site, site, "<init>", argTypes, typeParams, autoboxing);
+  }
+
   private Resolution findMethod(Env env, JavaType callSite, JavaType site, String name, List<JavaType> argTypes, List<JavaType> typeParams, boolean autoboxing) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "findMethod" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)